crates/ra_ide_api/src/impls.rs

```diff
@@ -0,0 +1,121 @@
+use ra_db::{SourceDatabase};
+use ra_syntax::{
+    AstNode, ast,
+    algo::find_node_at_offset,
+};
+use hir::{db::HirDatabase, source_binder};
+
+use crate::{FilePosition, NavigationTarget, db::RootDatabase, RangeInfo};
+
+pub(crate) fn goto_implementation(
+    db: &RootDatabase,
+    position: FilePosition,
+) -> Option<RangeInfo<Vec<NavigationTarget>>> {
+    let file = db.parse(position.file_id);
+    let syntax = file.syntax();
+
+    let krate_id = db.crate_for(position.file_id).pop()?;
+    let krate = hir::Crate { crate_id: krate_id };
```

Comment:
rather than making `crate_id` pub and constructing `Crate` by hand, we should bind the module first, and then use `module.krate()` method.